orcid-core/src/main/java/org/orcid/core/manager/impl/WorkManagerImpl.java

```diff
@@ -185,15 +185,15 @@ public Work createWork(String orcid, Work work, boolean applyAPIValidations) {
         
         if (applyAPIValidations) {
             activityValidator.validateWork(work, sourceEntity, true, applyAPIValidations, null);
-            Date lastModified = profileEntityManager.getLastModified(orcid);
-            long lastModifiedTime = (lastModified == null) ? 0 : lastModified.getTime();
-            List<MinimizedWorkEntity> works = workEntityCacheManager.retrieveMinimizedWorks(orcid, lastModifiedTime);
+            
             // If it is the user adding the peer review, allow him to add
             // duplicates
             if (!sourceEntity.getSourceId().equals(orcid)) {
-                if (works != null) {
-                    List<Work> workEntities = jpaJaxbWorkAdapter.toMinimizedWork(works);
-                    for (Work existing : workEntities) {
+                Date lastModified = profileEntityManager.getLastModified(orcid);
+                long lastModifiedTime = (lastModified == null) ? 0 : lastModified.getTime();
+                List<Work> existingWorks = this.findWorks(orcid, lastModifiedTime);       
```

Comment:
Hey @TomDemeranville did you tested this? I think the @cachable annotation is bypassed if the call is withing the same class: http://stackoverflow.com/questions/12115996/spring-cache-cacheable-method-ignored-when-called-from-within-the-same-class, thats why I didn't do this before.

Could you please give it a double check and confirm if the cache is used or not when called from the same class?

@rcpeters @wjrsimpson FYI

Thanks